mlflow/llama_index/__init__.py

```diff
@@ -348,32 +354,158 @@ def _load_pyfunc(path, model_config: Optional[Dict[str, Any]] = None):
+
+        if isinstance(data, str):
+            return QueryBundle.from_dict({"query_str": data})
+        elif isinstance(data, dict):
+            try:
+                return QueryBundle.from_dict(data)
+            except KeyError:
+                raise ValueError(
+                    "The input dictionary did not have the correct schema. It must support the "
+                    "supported in a llama_index QueryBundle: "
+                    "['query_str', 'image_path', 'custom_embedding_strs', 'embedding']"
+                )
+        elif isinstance(data, (list, np.ndarray)):
+            prediction_list = [self._format_predict_input_query_and_retriever(d) for d in data]
+        elif isinstance(data, pd.DataFrame):
+            prediction_list = [
+                self._format_predict_input_query_and_retriever(d)
+                for d in data.to_dict(orient="records")
+            ]
+        else:
+            raise ValueError(
+                f"Unsupported input type: {type(data)}. It must be one of "
+                "[str, dict, list, numpy.ndarray, pandas.DataFrame]"
+            )
+
+        if len(prediction_list) == 1:
+            return prediction_list[0]
+
+        return prediction_list
+
+    def _safe_convert_to_chat_messages(self, data) -> Union[List[ChatMessage], str]:
+        # NB: Chat messages that have been converted to a string via str() are formatted as
+        # "`role`: `content`", for example "system: You are a helpful bot".
+        if isinstance(data, (list, tuple)) and all(":" in message for message in data):
+            return [ChatMessage.from_str(*(message.split(": ", 1)[::-1])) for message in data]
+
+        return data
+
+    def _format_predict_input_chat(self, data):
+        """
+        Chat engines can have a variety of signatures. The two primary standards are:
+        1. chat(str, Sequence[ChatMessage])
+        2. chat(Sequence[ChatMessage])
+
+        There are not consistent naming conventions for either of these standards, so
+        we'll need to leverage positional arguments.
+        """
+        data = _convert_llm_input_data(data)
+
+        if isinstance(data, str):
+            return data
+        elif isinstance(data, dict):
+            if len(data) == 1:
+                # Assume that this value should be treated as a positional argument
+                payload = next(value for value in data.values())
+                return self._safe_convert_to_chat_messages(payload)
+
+            elif len(data) > 1:
+                # Assume that any kwarg value that's a sequence of strings with a colon
+                # should be converted to a list of ChatMessage objects.
+                return {k: self._safe_convert_to_chat_messages(v) for k, v in data.items()}
+        elif isinstance(data, (list, np.ndarray)):
+            prediction_list = [self._format_predict_input_chat(d) for d in data]
+        elif isinstance(data, pd.DataFrame):
```

Comment:
Can we check what are the possible output type from `_convert_llm_input_data`? From a quick read, seems `np.array` and `pd.DataFrame` are not returned.